### DISCUSSION TITLE
How to create a migration for identity schema changes?

### DISCUSSION THREAD

**Discussion by openscript** [ory/kratos#2178] [Q&A] [answered]
I've quite a simple change in an identity schema:

```
{
    "$id": "https://schemas.ory.sh/presets/kratos/quickstart/email-password/identity.schema.json",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "Employee",
    "type": "object",
    "properties": {
      "traits": {
        "type": "object",
        "properties": {
          "username": {
            "type": "string",
            "title": "Username",
            "minLength": 4,
            "maxLength": 36
          },
          "forename": {
            "type": "string",
            "title": "Forename",
            "minLength": 2,
            "maxLength": 36
          },
          "surname": {
            "type": "string",
            "title": "Surname",
            "minLength": 2,
            "maxLength": 36
          },
          "email": {
            "type": "string",
            "format": "email",
            "title": "E-Mail",
            "minLength": 3,
            "ory.sh/kratos": {
              "credentials": {
                "password": {
                  "identifier": true
                }
              },
              "verification": {
                "via": "email"
              },
              "recovery": {
                "via": "email"
              }
            }
          },
          "phone": {
            "type": "string",
            "title": "Phone",
            "maxLength": 18
          },
          "language": {
            "type": "string",
            "title": "Language"
          }
        },
        "required": [
          "username",
          "forename",
          "surname",
          "email",
          "phone"
        ],
        "additionalProperties": false
      }
    }
  }
  ```
  
  `phone` should be renamed to `mobile`. Is it somehow possible to manage such migrations with Kratos? I've used the documentation, but unfortunately some pages related to this don't work: https://www.ory.sh/kratos/docs/cli/kratos-migrate-sql/kratos-migrate which is linked here https://www.ory.sh/kratos/docs/cli/kratos-migrate-sql/#kratos-migrate-sql.
  
  Any help or idea is appreciated. Thank you :)

**Comment by aeneasr** [comment, marked as answer]
Hey there! Good question!

We do not have a native way of doing this right now (meaning an API you can call). What you can do though is to list the identities using the kratos cli `kratos identities list —format json` ( https://www.ory.sh/kratos/docs/cli/kratos-identities-list ) and then pipe that through something like jq (or some other tooling) and then use the `PUT /identities/:id` endpoint to update the user’s traits. The CLI does not yet support updating a user’s traits.

I would recommend to do some dry runs first and make back ups so nothing gets lost!

Our goal is to have a CLI task to assist with this, or, to speed things up an API endpoint

**Reply by openscript** [reply]
Hey @aeneasr!

Thank you for the quick response. I migrated the database directly with https://www.postgresql.org/docs/9.5/functions-json.html.

I know it's not nice, because the application which owns the database should have full control over it. Maybe an idea is to plug something like https://github.com/golang-migrate/migrate in?

I would be happy to contribute this to the project.

**Reply by aeneasr** [reply]
That makes sense, we want to provide a way to do this using an API or CLI, but don’t yet know how to do it efficiently when there‘s millions or billions of IDs. Any ideas are welcomed!
